lib/rspec/core/metadata_filter.rb

```diff
@@ -21,7 +21,7 @@ def filter_applies?(key, filter_value, metadata)
 
             meta_value = metadata.fetch(key) { return false }
 
-            return true if TrueClass === filter_value && !!meta_value
+            return true if TrueClass === filter_value && meta_value
```

Comment:
Because `filter_value` could be anything, and we can't count on it's `==` being defined properly.  We've repeatedly gotten bug reports from users when we've called `==` on their objects where they defined `==` to assume the passed object was of the same type, and tried to access specific attributes on it that did not exist.  Calling a method on `TrueClass` instead of `filter_value` avoids that issue.

It could be `true == filter_value` but `TrueClass ===` is in keeping with the style of the rest of the method which repeatedly uses `SomeClass === value`.